Release build fails: ModelChoice::ClaudeSonnet and ClaudeOpus not handled in UI routing

## Problem

The macOS .pkg installer release build fails with a non-exhaustive match error in `arkavo-cli`:

```
error[E0004]: non-exhaustive patterns: `ModelChoice::ClaudeSonnet` and `ModelChoice::ClaudeOpus` not covered
   --> crates/arkavo-cli/src/commands/ui.rs:531:11
```

## Root Cause

Commit 84583d38 added `ClaudeSonnet` and `ClaudeOpus` variants to the `ModelChoice` enum in `arkavo-router/src/decision.rs`, but the match statement in `crates/arkavo-cli/src/commands/ui.rs:531` (function `create_client_from_routing`) was not updated to handle these new variants.

## Impact

- Release workflow fails at the "Build macOS .pkg Installer" job
- Blocks new releases until fixed

## Solution

Add a match arm for `ModelChoice::ClaudeSonnet | ModelChoice::ClaudeOpus` that creates an `AnthropicProvider` client.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -95,7 +95,7 @@ exclude = ["crates/arkavo-protocol/fuzz"]
 resolver = "2"
 
 [workspace.package]
-version = "0.42.1"
+version = "0.42.2"
 edition = "2024"
 rust-version = "1.91.0"
 authors = ["Arkavo Contributors"]

--- a/crates/arkavo-cli/src/commands/ui.rs
+++ b/crates/arkavo-cli/src/commands/ui.rs
@@ -541,6 +541,18 @@ async fn create_client_from_routing(
                 Err("Gemini feature not enabled".into())
             }
         }
+        ModelChoice::ClaudeSonnet | ModelChoice::ClaudeOpus => {
+            #[cfg(feature = "llm-remote")]
+            {
+                use arkavo_llm::providers::AnthropicProvider;
+                let provider = Box::new(AnthropicProvider::from_env()?);
+                Ok(LlmClient::new(provider))
+            }
+            #[cfg(not(feature = "llm-remote"))]
+            {
+                Err("Anthropic support requires llm-remote feature".into())
+            }
+        }
         ModelChoice::LocalGemma270M
         | ModelChoice::LocalGemma4B
         | ModelChoice::LocalGemma12B

